client/app/pages/queries/components/QueryPageHeader.jsx

```diff
@@ -111,50 +114,46 @@ export default function QueryPageHeader({ query, sourceMode }) {
                 isDraft={query.is_draft}
                 isArchived={query.is_archived}
                 canEdit={query.can_edit}
-                getAvailableTags={loadTags}
+                getAvailableTags={getQueryTags}
                 onEdit={saveTags}
               />
             </span>
           </h3>
+          <span className="flex-fill" />
+          {query.is_draft && !query.isNew() && query.can_edit && (
+            <Button className="hidden-xs m-r-5" onClick={togglePublished}>
+              <i className="fa fa-paper-plane m-r-5" /> Publish
+            </Button>
+          )}
 
-          <span className="flex-fill">&nbsp;</span>
-
-          <Button.Group className="p-0 text-right text-nowrap align-self-start d-flex m-t-5">
-            {query.is_draft && !query.isNew() && query.can_edit && (
-              <Button onClick={togglePublished}>
-                <i className="fa fa-paper-plane m-r-5" /> Publish
-              </Button>
-            )}
-
-            {!query.isNew() && canViewSource && (
-              <span>
-                {!sourceMode && (
-                  <Button href={query.getUrl(true, selectedTab)}>
-                    <i className="fa fa-pencil-square-o m-r-5" aria-hidden="true" /> Edit Source
-                  </Button>
-                )}
-                {sourceMode && (
-                  <Button href={query.getUrl(false, selectedTab)} data-test="QueryPageShowDataOnly">
-                    <i className="fa fa-table m-r-5" aria-hidden="true" /> Show Data Only
-                  </Button>
-                )}
-              </span>
-            )}
+          {!query.isNew() && canViewSource && (
+            <span>
+              {!sourceMode && (
+                <Button className="m-r-5" href={query.getUrl(true, selectedTab)}>
+                  <i className="fa fa-pencil-square-o m-r-5" aria-hidden="true" /> Edit Source
+                </Button>
+              )}
+              {sourceMode && (
+                <Button className="m-r-5" href={query.getUrl(false, selectedTab)} data-test="QueryPageShowDataOnly">
+                  <i className="fa fa-table m-r-5" aria-hidden="true" /> Show Data Only
+                </Button>
+              )}
+            </span>
+          )}
 
-            {!query.isNew() && (
-              <Dropdown overlay={createMenu(moreActionsMenu)} trigger={['click']}>
-                <Button><i className="zmdi zmdi-more" /></Button>
-              </Dropdown>
-            )}
-          </Button.Group>
+          {!query.isNew() && (
+            <Dropdown overlay={createMenu(moreActionsMenu)} trigger={['click']}>
+              <Button><Icon type="ellipsis" rotate={90} /></Button>
+            </Dropdown>
+          )}
         </div>
-        <span className={cx('query-tags__mobile', { 'query-tags__empty': query.tags.length === 0 })}>
+        <span className={cx('m-t-5 query-tags__mobile', { 'query-tags__empty': query.tags.length === 0 })}>
```

Comment:
Seems it's better to remove padding at all for this component (update `.query-tags__mobile` rules) instead of using `m-t-5` - tags and query name should be closer on small screens.

Reply:
Old:

![image](https://user-images.githubusercontent.com/12139186/70820864-5428dd00-1de2-11ea-8908-ff31c4003f55.png)

This without fix:
![image](https://user-images.githubusercontent.com/12139186/70820898-686cda00-1de2-11ea-9e8f-d06d3d339730.png)

This with fix:
![image](https://user-images.githubusercontent.com/12139186/70820944-7d496d80-1de2-11ea-89b0-2e3aeabc3406.png)
![image](https://user-images.githubusercontent.com/12139186/70820967-8fc3a700-1de2-11ea-8141-7ccd3970e284.png)

Reply:
:+1: